### PR TITLE
Don't draw invisible polygons in the pre-roll

### PR DESCRIPTION
**Pull request by daschuer**
The issue has been discovered when scrolling far before the track. 
In this case there is a risk of a stalled GUI because of rendering thousand of polygons.

### CODE REVIEW

File: src/waveform/renderers/waveformrendererpreroll.cpp [hunk 1]
```diff
@@ -76,8 +76,14 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
...
             double triangleTipVSamplePosition =
-                    numberOfVSamples * playMarkerPositionFrac -
+                    numberOfVisibleVSamples * playMarkerPositionFrac -
                     currentVSamplePosition;
+            double invisibleVSamples = triangleTipVSamplePosition - numberOfVisibleVSamples;
+            if (invisibleVSamples > 0) {
+                triangleTipVSamplePosition -=
+                        floor(invisibleVSamples / polyVSampleOffset) *
+                        polyVSampleOffset;
+            }
```
Comment (ronso0) [inline src/waveform/renderers/waveformrendererpreroll.cpp:86]: Could you please add/refine some comments, for example
`// VSample position of the right-most triangle's tip just before the track start`
`// samples on the right of the viewport boundary`
`// get tip position of the last visible triangle to draw only visible triangles`
(pretty much the same, only adjusted wording, for the postroll area)

**Review by ronso0 — APPROVED** [review]
LGTM, just some more/better comments would be nice

File: src/waveform/renderers/waveformrendererpreroll.cpp [hunk 2]
```diff
@@ -100,11 +102,14 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
 
         if (postRollVisible) {
             const int remainingVSamples = totalVSamples - currentVSamplePosition;
-            // Sample position of the left-most triangle's tip
+            // VSample position of the left-most triangle's tip just after the track end
             double triangleTipVSamplePosition =
                     numberOfVisibleVSamples * playMarkerPositionFrac +
                     remainingVSamples;
+            // Number of invisible VSamples from the track start to the right of the
```
Comment (ronso0) [inline src/waveform/renderers/waveformrendererpreroll.cpp:109]:
```suggestion
            // Number of invisible VSamples from the track end to the right of the
```

**Comment by ronso0** [comment]
@daschuer Do you want to squash the commits?

**Comment by daschuer** [comment]
Yes, done.

**Comment by ronso0** [comment]
Great, thanks

**Pull request merged**